common/types.go

```diff
@@ -187,4 +188,6 @@ type BaseComponent interface {
 	GetRoute() BaseComponentRoute
 	GetBindings() BaseComponentBindings
 	GetAffinity() BaseComponentAffinity
+	GetDeploymentStrategy() *appsv1.DeploymentStrategy
```

Comment:
We'll likely have more parameters under deployment and statefulSet in the future. It'll be good to create these update strategies under another level - similar to how GetService() BaseComponentService is setup. Then BaseComponentService has various methods (i.e. GetPort() int32)